clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -334,23 +338,27 @@ public boolean isInternal() {
 
     }
 
+    // TODO: rename this class to MetadataResponsePartitionState
```

Comment:
`MetadataResponsePartitionState` will be more consistent and clearer than `PartitionMetadata` in describing the usage of this class. It is not necessary to change the class. I replaced the TODO with a comment that says `This is used to describe MetadataResponsePartitionState`.